internal/controllers/topology/cluster/patches/template.go

```diff
@@ -56,10 +57,12 @@ func newRequestItemBuilder(template *unstructured.Unstructured) *requestItemBuil
 }
 
 // WithHolder adds holder to the requestItemBuilder.
-func (t *requestItemBuilder) WithHolder(object client.Object, fieldPath string) *requestItemBuilder {
+// Note: We pass in gvk explicitly as we can't rely on GVK being set on all objects
+// (only on Unstructured).
+func (t *requestItemBuilder) WithHolder(object client.Object, gvk schema.GroupVersionKind, fieldPath string) *requestItemBuilder {
```

Comment:
What about
```suggestion
func (t *requestItemBuilder) WithHolder(gvk schema.GroupVersionKind, object client.Object, fieldPath string) *requestItemBuilder {
```

Reply:
Similar to above. But I'm okay with both variants